app/services/streaming/streaming.ts

```diff
@@ -1,261 +1,221 @@
+import { StatefulService, mutation } from 'services/stateful-service';
+import { ObsApiService, EOutputCode } from 'services/obs-api';
+import { Inject } from 'util/injector';
 import moment from 'moment';
-
-import { Inject } from '../../util/injector';
-import { StatefulService, mutation } from '../stateful-service';
-import { nodeObs } from '../obs-api';
-import { SettingsService } from '../settings';
-import { padStart, isEqual } from 'lodash';
-import { track } from '../usage-statistics';
-import { WindowsService } from '../windows';
+import { padStart } from 'lodash';
+import { SettingsService } from 'services/settings';
+import { WindowsService } from 'services/windows';
 import { Subject } from 'rxjs/Subject';
-import { IStreamingServiceApi, IStreamingServiceState } from './streaming-api';
 import electron from 'electron';
+import {
+  IStreamingServiceApi,
+  IStreamingServiceState,
+  EStreamingState,
+  ERecordingState
+} from './streaming-api';
+
+enum EOBSOutputType {
+  Streaming = 'streaming',
+  Recording = 'recording'
+}
 
+enum EOBSOutputSignal {
+  Starting = 'starting',
+  Start = 'start',
+  Stopping = 'stopping',
+  Stop = 'stop'
+}
 
-export class StreamingService extends StatefulService<IStreamingServiceState> implements IStreamingServiceApi {
+interface IOBSOutputSignalInfo {
+  type: EOBSOutputType;
+  signal: EOBSOutputSignal;
+  code: EOutputCode;
+  error: string;
+}
 
+export class StreamingService extends StatefulService<IStreamingServiceState>
+  implements IStreamingServiceApi {
+  @Inject() obsApiService: ObsApiService;
   @Inject() settingsService: SettingsService;
   @Inject() windowsService: WindowsService;
 
-  static initialState = {
-    isStreaming: false, // Doesn't account for delay
-    isLive: false, // Accounts for delay
-    usingDelay: false,
-    streamStartTime: null,
-    streamEndTime: null,
-    isRecording: false,
-    recordStartTime: null,
-    streamOk: null
-  } as IStreamingServiceState;
+  streamingStatusChange = new Subject<EStreamingState>();
 
-  @mutation()
-  START_STREAMING(startTime: string, usingDelay: boolean) {
-    this.state.isStreaming = true;
-    this.state.streamStartTime = startTime;
-    this.state.streamEndTime = null;
-    this.state.usingDelay = usingDelay;
-  }
+  // Dummy subscription for stream deck
+  streamingStateChange = new Subject<void>();
```

Comment:
If we don't emit this event now, is it just for preventing errors with streamdeck?

Reply:
Yeah basically.  I think stream deck probably ignores the error, but I wanted to make sure there weren't any issues with subscribing.  It turns out they're not actually using the event for anything anyway.